instrumentation/httpurlconnection/library/src/main/java/io/opentelemetry/instrumentation/library/httpurlconnection/HttpUrlReplacements.java

```diff
@@ -0,0 +1,402 @@
+            HttpURLConnection c, String name, long Default) {
+        // URLConnection also overrides this and that is covered in replacementForHeaderFieldDate
+        // method.
+        return replace(c, () -> c.getHeaderFieldDate(name, Default));
+    }
+
+    public static String replacementForHeaderFieldKey(URLConnection c, int n) {
+        // HttpURLConnection also overrides this and that is covered in
+        // replacementForHttpHeaderFieldKey method.
+        return replace(c, () -> c.getHeaderFieldKey(n));
+    }
+
+    public static String replacementForHttpHeaderFieldKey(HttpURLConnection c, int n) {
+        // URLConnection also overrides this and that is covered in replacementForHeaderFieldKey
+        // method.
+        return replace(c, () -> c.getHeaderFieldKey(n));
+    }
+
+    public static String replacementForHeaderField(URLConnection c, int n) {
+        // HttpURLConnection also overrides this and that is covered in
+        // replacementForHttpHeaderField method.
+        return replace(c, () -> c.getHeaderField(n));
+    }
+
+    public static String replacementForHttpHeaderField(HttpURLConnection c, int n) {
+        // URLConnection also overrides this and that is covered in replacementForHeaderField
+        // method.
+        return replace(c, () -> c.getHeaderField(n));
+    }
+
+    public static int replacementForResponseCode(URLConnection c) throws IOException {
+        HttpURLConnection httpURLConnection = (HttpURLConnection) c;
+        return replaceThrowable(c, httpURLConnection::getResponseCode);
+    }
+
+    public static String replacementForResponseMessage(URLConnection c) throws IOException {
+        HttpURLConnection httpURLConnection = (HttpURLConnection) c;
+        return replaceThrowable(c, httpURLConnection::getResponseMessage);
+    }
+
+    public static OutputStream replacementForOutputStream(URLConnection c) throws IOException {
+        return replaceThrowable(c, c::getOutputStream, false);
+    }
+
+    public static InputStream replacementForInputStream(URLConnection c) throws IOException {
+        startTracingAtFirstConnection(c);
+
+        InputStream inputStream;
+        try {
+            inputStream = c.getInputStream();
+        } catch (IOException e) {
+            reportWithThrowable(c, e);
+            throw e;
+        }
+
+        if (inputStream == null) {
+            return inputStream;
+        }
+
+        return getInstrumentedInputStream(c, inputStream);
+    }
+
+    public static InputStream replacementForErrorStream(HttpURLConnection c) {
+        startTracingAtFirstConnection(c);
+
+        InputStream errorStream = c.getErrorStream();
+
+        if (errorStream == null) {
+            return errorStream;
+        }
+
+        return getInstrumentedInputStream(c, errorStream);
+    }
+
+    private static InputStream getInstrumentedInputStream(
+            URLConnection c, InputStream inputStream) {
+        return new InputStream() {
```

Comment:
Might be slightly more readable to not have this inline, but pulled out to an inner class.